File: problems/006/006.js
function f() {
    var i, sum = 0, sumOfSquares = 0;

    for (i = 1; i < 101; i++) {
        sum += i;
        sumOfSquares += i * i;
    }

    var squaredSum = sum * sum;

    return squaredSum - sumOfSquares;
}

function toNs(ts) {
    return ts.sec * 1000000000 + ts.nsec;
}

var clock = require('posix-clock');

if (process.argv.length === 3) {
    var i;
    var iters = parseInt(process.argv[2], 10);

    var start = clock.gettime(clock.MONOTONIC);
    for (i = 0; i < iters; i++) {
        f();
    }
    var end = clock.gettime(clock.MONOTONIC);

    console.log(toNs(end) - toNs(start));
} else {
    console.log(f());
}

File: problems/006/006.lua
posix = require "posix"

local function f()
    local sum_of_squares, sum = 0, 0;

    for i = 1,100 do
        sum = sum + i;
        sum_of_squares = sum_of_squares + i * i;
    end

    local squared_sum = sum * sum;

    return squared_sum - sum_of_squares;
end

function toNs(ts)
    return ts[1] * 1000000000 + ts[2]
end

if #arg == 0 then
    print(f())
elseif #arg == 1 then
    iters = tonumber(arg[1])
    start = { posix.clock_gettime("monotonic") }
    for i = 1,iters do
        f()
    end
    end_ = { posix.clock_gettime("monotonic") }
    print(toNs(end_) - toNs(start))
end
